app/app.js

```diff
@@ -1,190 +1,23 @@
-})
-
-router.post('/test-send-many', function (req, res) {
-  setCampaign();
-  sendToMany();
-	res.send('"How was your week?" - notifications sent to private channels. Thanks!');
-})
-
-router.post('/test-read', function(req, res) {
-  console.log('TEST READ')
-  res.send('OK')
-
-  // return firebase.database().ref('/votes').once('value').then(function(snapshot) {
-  //   var collections = snapshot.val();
-  //   readVotes(collections);
-  // }).then(function() {
-  //   res.send('All ok');
-  // });
-})
-// </Routers>
-
-// <Functions>
-function readVotes(votes) {
-  for (var campaign in votes) {
-    var dateNow = parseInt(campaign.substring(1, campaign.lenght));
-    date = new Date(dateNow);
-    var dateOfCampaign = date.toDateString();
-
-    console.log('Campaign date : ' + dateOfCampaign);
-    campaignCollection = votes[campaign];
-    for (var userId in campaignCollection) {
-      console.log(slackUsers[userId] + ' ' + campaignCollection[userId].mood);
-    }
-  }
-}
-
-function getUsersList() {
-  slack = new Slack(configResolver.getConfigVariable('API_TOKEN'));
-  slack.api(listUsersMethod, {
-  }, function(err, response){
-
-    for(var index in response.members) {
-      if(!response.members[index].is_bot && !response.members[index].deleted) {
-        slackUsers[response.members[index].id] = response.members[index].name;
-      }
-    }
-  })
-}
-
-function signInWithEmailPass(email, password) {
-  firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
-    var errorCode = error.code;
-    var errorMessage = error.message;
-  })
-}
-
-function setCampaign() {
-  campaignId = 'C' + Date.now();
-}
-
-function getUrlForRequest(username) {
-  return 'https://slack.com/api/' + postMessageMethod +
-    '?token=' + configResolver.getConfigVariable('API_TOKEN') +
-    '&username=Mr. Moody' +
-    '&as_user=false' +
-    '&icon_url=https://s3-us-west-2.amazonaws.com/slack-files2/avatars/2016-11-04/100929399430_30f602e36ebfbc81756b_48.jpg' +
-    '&channel=@' + username +
-    '&attachments=' + getAttachments();
-}
-
-function makeHttpsGetRequest(url) {
-  https.get(url, (res) => {
-  }).on('error', (e) => {
-    console.error('HTTP error:', e);
-  });
-
-  return true;
-}
-
-function sendToMany() {
-  slack = new Slack(configResolver.getConfigVariable('API_TOKEN'));
-  slack.api(listUsersMethod, {
-  }, function(err, response){
-
-    for(var index in response.members){
-      if(!response.members[index].is_bot && !response.members[index].deleted) {
-        slackUsername = response.members[index].name;
-        var url = getUrlForRequest(slackUsername);
-        makeHttpsGetRequest(url);
-      }
-    }
-  });
-
-  return false;
-}
+const configResolver = require('./modules/configResolver.js')
+const version = 'v.0.0.12'
 
+app.use(bodyParser.urlencoded({ extended: true }))
+app.use(bodyParser.json())
 
+const mainController = require('./controllers/mainController')(router)
```

Comment:
this is more a personal opinion to have things in a more organized structure you could create a new router instead of sending this one to register routes under it
something like the following
```js
const router = require('./controllers'); // should the path be called routers ?
router.use('/api', router);
```
`controllers/index.js`
```js
const router = new express.Router()

router.use(validationMiddleware.validateRequest)
router.use(errorMiddleware.errorMiddleware)

// be aware that previously registered middleware will run before any router registered after
router.use(slackCommands)
router.use(campaigns)

module.exports = router;
```
``

`controllers/campaignsController.js`
```js
const router = new express.Router()

// router.get
// router.post
// ...

module.exports = router;
```